visualization/script/build-nwjs.js

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const zip = require('bestzip');
+const path = require('node:path');
+
+fs.rmSync('./build', { recursive: true, force: true });
+fs.rmSync('dist/packages', { recursive: true, force: true });
+
+const NwBuilder = require('nw-builder');
+const nw = new NwBuilder({
+  buildDir: "dist/packages",
+  cacheDir: "build",
+  forceDownload: true,
+  files: [
+    "./dist/webpack/**/*",
+    "package.json"
+  ],
+  flavor: "normal",
+  version: "latest",
+  platforms: ["osx64", "win32", "win64", "linux32", "linux64"],
+  appName: null,
+  appVersion: null,
+  buildType: function () { return this.appName + "-" + this.appVersion },
+  useRcedit: true,
+  zip: false,
+  macIcns: './app/assets/icon.icns',
+  winIco:'./app/assets/icon.ico',
+  winVersionString: {
+    'CompanyName': 'MaibornWolff GmbH',
+    'FileDescription': 'CodeCharta Visualization',
+    'ProductName': 'codecharta-visualization',
+    'LegalCopyright': 'Copyright MaibornWolff GmbH'
+  }
+});
+nw.build().then(function () {
+  console.log("nw-build done!");
+  fs.mkdirSync(nw.options.buildDir, { recursive: true });
+  const baseZipPath = path.resolve(nw.options.buildDir, nw.options.appName + "-" + nw.options.appVersion + "-");
+  nw._forEachPlatform(function (name, platform) {
+    zip({
+      source:name,
+      destination: baseZipPath + name + ".zip",
+      cwd: path.join(platform.releasePath, "..")
+    }).then(function () {
+      console.log(name + " zipped!")
+    }).catch(function (err) {
+      console.error(err.stack)
+      process.exit(1)
+    });
+  });
+
+  zip({
```

Comment:
You use two run away promises here. Instead, you could just write:

```js
await Promise.all([
  zip({ ... }).
  zip({ ... }),
])
console.log('web and %s zipped', name)
```

Do not use explicit catch clauses. The error is going to be logged anyway.
Only logging the stack is also going to potantially hide important information of an error. Instead, always log the error. The stack is just a string that is a combination of `error.name + error.message + error.frames`. An error could include many other things as well such as a cause.